nodes/web/js/launcher.js

```diff
@@ -0,0 +1,309 @@
+            method: 'POST',
+            headers: { 
+                'Content-Type': 'application/json',
+                'Accept': 'application/json'
+            },
+            body: JSON.stringify({ 
+                action,
+                settings
+            })
+        });
+        
+        if (!response.ok) {
+            // Reset transitional states on error
+            if (action === 'start') {
+                updateStatusIndicator({ starting: false });
+            } else if (action === 'stop' || action === 'restart') {
+                updateStatusIndicator({ stopping: false });
+            }
+            
+            const errorText = await response.text();
+            console.error("[ComfyStream] Server returned error response:", response.status, errorText);
+            try {
+                const errorData = JSON.parse(errorText);
+                throw new Error(errorData.error || `Server error: ${response.status}`);
+            } catch (e) {
+                throw new Error(`Server error: ${response.status} - ${errorText}`);
+            }
+        }
+
+        const data = await response.json();
+        
+        // Update status indicator after control action
+        if (data.status) {
+            // Clear transitional states
+            data.status.starting = false;
+            data.status.stopping = false;
+            updateStatusIndicator(data.status);
+        }
+        
+        return data;
+    } catch (error) {
+        // Reset any transitional states on error
+        if (action === 'start') {
+            updateStatusIndicator({ starting: false });
+        } else if (action === 'stop' || action === 'restart') {
+            updateStatusIndicator({ stopping: false });
+        }
+        
+        console.error('[ComfyStream] Error controlling server:', error);
+        app.ui.dialog.show('Error', error.message || 'Failed to control ComfyStream server');
+        throw error;
+    }
+}
+
+async function openUI() {
+    try {
+        // Get settings from the settings manager
+        const settings = settingsManager.getCurrentHostPort();
+        
+        const response = await fetch('/launch_comfystream', {
+            method: 'POST',
+            headers: { 
+                'Content-Type': 'application/json',
+                'Accept': 'application/json'
+            },
+            body: JSON.stringify({ settings })
+        });
+        
+        if (!response.ok) {
+            const errorText = await response.text();
+            console.error("[ComfyStream] UI launch returned error response:", response.status, errorText);
+            try {
+                const errorData = JSON.parse(errorText);
+                throw new Error(errorData.error || `Server error: ${response.status}`);
+            } catch (e) {
+                throw new Error(`Server error: ${response.status} - ${errorText}`);
+            }
+        }
+
+        const data = await response.json();
```

Comment:
In my testing the button doesn't work very reliably. This prevents a potential open redirect vulnerability and improves the reliability of the **Open ComfyStream UI** button. 
```suggestion
        const data = await response.json();
        data.url = "./extensions/comfystream/static/index.html"
```